src/connection.ts

```diff
@@ -59,21 +60,21 @@ type Constructor<
  */
 export class StableWSConnection<
   ChannelType extends UnknownType = UnknownType,
-  UserType extends UnknownType = UnknownType,
-  CommandType extends string = LiteralStringForUnion
+  CommandType extends string = LiteralStringForUnion,
+  UserType extends UnknownType = UnknownType
 > {
-  wsBaseURL: Constructor<ChannelType, UserType, CommandType>['wsBaseURL'];
-  clientID: Constructor<ChannelType, UserType, CommandType>['clientID'];
-  userID: Constructor<ChannelType, UserType, CommandType>['userID'];
-  user: Constructor<ChannelType, UserType, CommandType>['user'];
-  userAgent: Constructor<ChannelType, UserType, CommandType>['userAgent'];
-  authType: Constructor<ChannelType, UserType, CommandType>['authType'];
-  apiKey: Constructor<ChannelType, UserType, CommandType>['apiKey'];
-  tokenManager: Constructor<ChannelType, UserType, CommandType>['tokenManager'];
-  messageCallback: Constructor<ChannelType, UserType, CommandType>['messageCallback'];
-  recoverCallback: Constructor<ChannelType, UserType, CommandType>['recoverCallback'];
-  eventCallback: Constructor<ChannelType, UserType, CommandType>['eventCallback'];
-  logger: Constructor<ChannelType, UserType, CommandType>['logger'];
+  wsBaseURL: Constructor<ChannelType, CommandType, UserType>['wsBaseURL'];
+  clientID: Constructor<ChannelType, CommandType, UserType>['clientID'];
+  userID: Constructor<ChannelType, CommandType, UserType>['userID'];
+  user: Constructor<ChannelType, CommandType, UserType>['user'];
+  userAgent: Constructor<ChannelType, CommandType, UserType>['userAgent'];
+  authType: Constructor<ChannelType, CommandType, UserType>['authType'];
+  apiKey: Constructor<ChannelType, CommandType, UserType>['apiKey'];
+  tokenManager: Constructor<ChannelType, CommandType, UserType>['tokenManager'];
+  messageCallback: Constructor<ChannelType, CommandType, UserType>['messageCallback'];
+  recoverCallback: Constructor<ChannelType, CommandType, UserType>['recoverCallback'];
+  eventCallback: Constructor<ChannelType, CommandType, UserType>['eventCallback'];
+  logger: Constructor<ChannelType, CommandType, UserType>['logger'];
```

Comment:
these are in usefulness order?

Reply:
we were mostly abcing the generics where it effected use, did these for you though.